03/app.js

```diff
@@ -3,22 +3,60 @@ import ReactDOM from 'react-dom';
 
 import Category from './Category';
 import Cart from './Cart';
+import Product from './Product';
 
 import data from './data.json';
 
 class App extends React.Component {
     state = {
         cart: [],
     }
+
     
     render() {
+        const {cart} = this.state;
         return (
             <section>
-                <Category />
-                <Cart />
+                <Category>
+                    {this.displayCategoryProducts(data)}
+                </Category>
+                <Cart>
+                    {this.displayCartProducts(cart)}
+                </Cart>
             </section>
         )
     }
+
+    displayCategoryProducts(data) {
+        return data.map(item => <Product name={item.name} price={item.price} buttonText={'Dodaj do koszyka'} onClick={this.addItem} />)
```

Comment:
Jeśli nie chcesz operować na DOM to powinieneś przekazywać `id`, które pozwoli Ci sprawdzić jaki jest to element tj.
`onClick={ () => this.addItem(item.id) }`  możesz też parametr przekazać cały obiekt ;)